example/app/code/local/MyVendor/MyModule/Models/Adapters/ConfigAdapter.php

```diff
@@ -0,0 +1,14 @@
+<?php 
+
+class MyVendor_MyModule_Models_Adapters_ConfigAdapter
+{
+    /**
+     * Return Magento Base Directory
+     *
+     * @return string
+     */
+    public function getBaseDir()
```

Comment:
This doesn't need to match the static method name. If the method was called `getBaseDirectory` it would be more readable and address Ciaran's comment about API translation.